pkg/storage/unified/resource/keys.go

```diff
@@ -5,6 +5,7 @@ import (
 	"strings"
 
 	"github.com/grafana/grafana/pkg/storage/unified/resourcepb"
+	"github.com/grafana/grafana/pkg/util"
 )
 
 func verifyRequestKey(key *resourcepb.ResourceKey) *resourcepb.ErrorResult {
```

Comment:
It would be great to add more tests to `keys_test.go` mostly as a way to see what valid and invalid looks like.

and add a case to make sure long names still work (but i don't think group + namespace or resource should be too long)